report/5.Conclusion/Discussion.tex

```diff
@@ -1,4 +1,62 @@
-% !TeX root = ../main.tex
-\chapter{Discussion}
-
+% !TeX root = ../main.tex

+\chapter{Discussion}
+\section{Discussion}
+In this chapter, the project is reflected upon.
+This will include the choices made in regards to the developed solution and report, as well as how the group delegated work to the individual group members and assured the quality of said work.
+Furthermore the major choices, not already touched upon are discussed in this chapter.
+
+The goal of HCL is to develop an "english-like" language.
+The choice of an english-like language, instead of a danish-like language, is to easier introduce programming concepts to the target-group.
+By being an english-like language, HCL expressions are designed to naturally resemble english written sentences.
+If proper naming conventions are followed, the written code becomes more intuitive.
+As an example snippet \ref{lis:discussionintuitive} shows why this is the case.
+
+\begin{lstlisting}[language=HCL,label=lis:discussionintuitive,firstnumber=1,caption=Example of the intuitive nature of HCL]
+func doSomething = () : none { Some Code that does stuff }
+func thisIsTrue = () : bool { Some code that returns a bool }
+
+:doSomething while :thisIsTrue
+\end{lstlisting}
+
+As discussed in the problem analysis, the majority of the target group have no understanding of operators.
+Users of HCL do not need to know the basics of logic operations as presented in discrete mathematics.
+This is because the implicit understanding of logical constructs, is natural in HCL, as the built-in operators have english names, instead of just being a character.
+As is evident by the statements made by the gymnasium students, novices would not be able to understand the logic operation shown below.
+\begin{center}
+	$(P \wedge Q) \vee R$
+\end{center}
+Naturally, because they would not be familiar with the precedence or association of the logical operators.
+In HCL this becomes much simpler, since all function calls are left associative and no operators exists.
+The snippet \ref{lis:discussionoperatorIntuitiveness} shows the equivalent HCL code.
+\begin{lstlisting}[language=HCL,label=lis:discussionoperatorIntuitiveness,firstnumber=1,caption=Example of the intuitive nature of HCL]
+(P and Q) or R
+\end{lstlisting}
+Types in HCL are immutable.
+This is because in algebra, when x is assigned the value 4, x simply holds the value of 4, with x being treated as a constant.

```

Comment:
But but but... Our types are immutable, but variables are mutable so in HCL if x is assigned the value 4, nothing prevents me from reassigning it to 7. So very bad (directly worng) example :stuck_out_tongue:

Reply:
Well this is retarded. In defense, I was hungry.